cost/azure/rightsize_managed_disks/azure_rightsize_managed_disks.pt

```diff
@@ -0,0 +1,1135 @@
+      resourceID: disk.id,
+      tags: tags,
+      resourceType: currentDiskName,
+      resourceTypeRedundancy: currentDiskRedundancy,
+      resourceTypeGiB: currentDiskData.sizeGiB,
+      resourceTypeIOPS: currentDiskData.IOPS,
+      resourceTypeThroughput: currentDiskData.throughput,
+      resourceTypeMonthlyPrice: currentDiskPrice.pricePerUnit,
+      resourceTypePriceCurrency: currentDiskPrice.currencyCode,
+      newResourceType: newResourceType,
+      newResourceTypeRedundancy: newResourceTypeRedundancy,
+      newResourceTypeGiB: newResourceTypeGiB,
+      newResourceTypeIOPS: newResourceTypeIOPS,
+      newResourceTypeThroughput: newResourceTypeThroughput,
+      newResourceTypeMonthlyPrice: newResourceTypeMonthlyPrice,
+      newResourceTypePriceCurrency: newResourceTypePriceCurrency,
+      resourceKind: disk.resourceType,
+      region: disk.region,
+      monthlySavings: monthlySavings.toFixed(2),
+      savingsCurrency: currentDiskPrice.currencyCode,
+      iopsPctMaximum: Math.round(disk.iopsConsumedPctMax),
+      iopsPctAverage: Math.round(disk.iopsConsumedPctAvg),
+      iopsThreshold: param_min_used_disk_iops_pct,
+      throughputPctMaximum: Math.round(disk.bandwithConsumedPctMax),
+      throughputPctAverage: Math.round(disk.bandwithConsumedPctAvg),
+      throughputThreshold: param_min_used_disk_throughput_pct,
+      sizeGB: disk.diskSizeGB,
+      usedGBPct: disk.diskUsedGB != undefined ? Math.round(disk.diskUsedGB * 100 / disk.diskSizeGB) : "Unknown",
+      diskSpaceThreshold: ds_min_used_disk_space_pct,
+      lookbackPeriod: param_stats_lookback,
+      service: disk.service.split("/")[0],
+    })
+
+    result.monthlySavings += monthlySavings
+  }
+
+  if (result.monthlySavings >= param_min_savings) {
+    result.monthlySavings = result.monthlySavings.toFixed(2)
+    result.raiseIncident = true
+  }
+EOS
+end
+
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_azure_rightsize_managed_disks" do
+  validate $ds_disk_rightsizing_recommendations do
+    summary_template "{{ data.policyName }}: There are {{ len data.recommendations }} oversized Azure Managed Disks, you could save ${{ data.monthlySavings }} USD!"
+    detail_template <<-'EOS'
+## Summary
+Found {{ len data.recommendations }} Azure Managed Disks that are *oversized* during the last {{ parameters.param_stats_lookback }} days. *Oversized* disks are those which have **less or equal than {{ parameters.param_min_used_disk_iops_pct }}% {{ parameters.param_stats_aggregation_for_disk_iops }} IOPS utilization, less or equal than {{ parameters.param_min_used_disk_throughput_pct }}% {{ parameters.param_stats_aggregation_for_disk_throughput }} throughput utilization and less or equal than {{ parameters.ds_min_used_disk_space_pct }}% used capacity (GiB)** within the last {{ parameters.param_stats_lookback }} days.
```

Comment:
This should be formatted to look like the other recommendations policies and the message should largely be constructed in the final datasource. For example, from the Azure Unused Volumes policy:

    detail_template <<-'EOS'
    **Potential Monthly Savings:** {{ with index data 0 }}{{ .total_savings }}{{ end }}

    {{ with index data 0 }}{{ .message }}{{ end }}
    EOS

Reply:
Thanks for this comment!

I addressed this by constructing most of the message in the previous `datasource`, in this case I'm using strictly USD for the currency since all the APIs this policy is working with only use dollars. I also tried to simplify the logic of the message construction and improve the readability for the user putting in separate points what were the parameters he specified.

Reply:
![image](https://github.com/flexera-public/policy_templates/assets/54189123/252b29c8-bd2c-4245-b216-7eb3b0e54f44)
![image](https://github.com/flexera-public/policy_templates/assets/54189123/83263973-cf54-4be6-a1f8-570a1fb7eb2a)